metadata/errors.go

```diff
@@ -1,22 +1,117 @@
 package metadata
 
-import "github.com/pkg/errors"
+type metadataExistsErr struct {
+	desc string
+}
+type metadataNotFoundErr struct {
+	desc string
+}
+type metadataNotEmptyErr struct {
+	desc string
+}
+
+// ErrExists is returned when an item already exists in metadata
+func ErrExists(msg string) error {
```

Comment:
Per the comment above, would it make sense for this to be `ErrExists(cause error) error`?